File: Client/app/typescript/module/season/SeasonViewModel.ts
module FifaLeagueClient.Module.Season {

    // Class showing a season with all the leagues
    export class SeasonViewModel
    {
         public Id:number;
         public Name:string;
         public LeagueViewModels:League.LeagueViewModel[];

        // build the model directly from the data returned by the service
        constructor(data){
            if(data != null){
                this.Id = data.Id;
                this.Name = data.Name;
                this.LeagueViewModels = data.LeagueViewModels;
            }
        }
    }

}

File: Server/FIFA.Server/Models/Season/SeasonViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace FIFA.Server.Models
{

    // Class the season name with all it s leagues
    public class SeasonViewModel
    {
         public int Id { get; set; }
         public string Name { get; set; }
         public IEnumerable<LeagueViewModel> LeagueViewModels { get; set; }
    }
}